src/main/main.ts

```diff
@@ -82,33 +82,39 @@ process.env.ELECTRON_DISABLE_SECURITY_WARNINGS = 'true';
 
 // Check for update
 const checkUpdateOnStartup = localStorage.getItem('check-upd-on-strtup') === 'true';
-if (app.isPackaged && checkUpdateOnStartup) {
-    hasUpdate(version)
-        .then(async (latest) => {
-            if (!latest) return;
-
-            const splitBody = latest.body.split('\n');
-            const changelogItems = splitBody.filter(
-                (line) => line.startsWith('- ') || line.startsWith('* ') || line.startsWith('• ')
-            );
-
-            const buttonResult = await dialog.showMessageBox(BrowserWindow.getFocusedWindow()!, {
-                type: 'info',
-                title: 'An update is available for chaiNNer!',
-                message: `Version ${latest.version} is available for download from GitHub.`,
-                detail: `Currently installed: ${version}\n\nRelease notes:\n\n${changelogItems.join(
-                    '\n'
-                )}`,
-                buttons: [`Get version ${latest.version}`, 'Ok'],
-                defaultId: 1,
-            });
-            if (buttonResult.response === 0) {
-                await shell.openExternal(latest.releaseUrl);
-                app.exit();
-            }
-        })
-        .catch((reason) => log.error(reason));
-}
+const checkForUpdate = () => {
+    if (app.isPackaged && checkUpdateOnStartup) {
```

Comment:
The name `checkForUpdate` does not imply that it only checks on start up. Could you move the `checkUpdateOnStartup` check to where the function is called?